exit/test_exit.c

```diff
@@ -0,0 +1,513 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    POSIX.1-2017 standard library functions tests
+ *    HEADER:
+ *    - stdlib.h
+ *    - unistd.h
+ *    TESTED:
+ *    - exit()
+ *    - _exit() (_Exit equivalent)
+ *    - atexit()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <unity_fixture.h>
+
+#define PATH1    "/tmp/testFile1"
+#define PATH2    "/tmp/testFile2"
+#define PATH3    "/tmp/testFile3"
+#define PATH4    "/tmp/testFile4"
+#define TEST_STR "test123"
+
+/* Set to random value */
+static int test_creatRes = 0x12345678;
+
+/* SIGCHLD signal handler */
+static void test_handler(int signum)
+{
+	/* Create file to confirm that handler has been invoked */
+	test_creatRes = creat(PATH1, DEFFILEMODE);
+}
+
+
+/* Functions to be registered by atexit */
+static void test_fun1(void)
+{
+	int fd, val = 1;
+
+	fd = open(PATH2, O_WRONLY | O_APPEND);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+	TEST_ASSERT_EQUAL_INT(sizeof(int), write(fd, &val, sizeof(int)));
+	close(fd);
+}
+
+
+static void test_fun2(void)
+{
+	int fd, val = 1234;
+
+	fd = open(PATH2, O_WRONLY | O_APPEND);
+	TEST_ASSERT_NOT_EQUAL_INT(-1, fd);
+	TEST_ASSERT_EQUAL_INT(sizeof(int), write(fd, &val, sizeof(int)));
+	close(fd);
+}
+
+
+TEST_GROUP(_exit);
+
+
+TEST_SETUP(_exit)
+{
+}
+
+
+TEST_TEAR_DOWN(_exit)
+{
+}
+
+
+TEST(_exit, status_vals)
+{
+	/* Return all possible least significant byte values and check whether wait() works */
+	pid_t pid;
+	int i;
+	int val[3];
+	/*
+	 * Only the least significant 8 bits shall be available to a waiting parent process,
+	 * 3 most significant bytes should be cut off so returning val[0] should resolve to returning 0, val[1] to 1 and val[2] to 2
+	 */
+	val[0] = 0x1 << 8;
+	val[1] = (0x1 << 16) + 1;
+	val[2] = (0x1 << 24) + 2;
+
+	for (i = 0; i < 256; i++) {
+		pid = fork();
+		TEST_ASSERT_GREATER_OR_EQUAL(0, pid);
+		/* child */
+		if (pid == 0) {
+			_exit(i);
+		}
+		/* parent */
+		else {
+			int status, ret;
+
+			ret = wait(&status);
+			TEST_ASSERT_EQUAL_INT(pid, ret);
+			TEST_ASSERT_EQUAL_INT(i, WEXITSTATUS(status));
+		}
+	}
+	/* Some exceeding values */
+	for (i = 0; i < 3; i++) {
+		pid = fork();
+		if (pid < 0) {
+			TEST_ASSERT_LESS_THAN_INT(0, pid);
+		}
```

Comment:
Such `if`s were pointed in previous review. All are corrected  so i think this flew unchanged under the radar :)